lib/resValidator.js

```diff
@@ -111,7 +111,16 @@ function validateResAgainstApiSpec(res, expectedResponse, openApiSpec) {
 
 
 function getOAPath(req, openApiSpec) {
-  const pathname = url.parse(req.path).pathname; // excludes the query (because: path = pathname + query)
+  let pathname = url.parse(req.path).pathname; // excludes the query (because: path = pathname + query)
```

Comment:
nit: would it make sense to move these 8 lines that you've added/changed be abstracted into `findOAPathMatchingPathname`? (I'm not sure)

Reply:
Abstracting that also into the `findOAPathMatchingPathname` function means we would be passing in the `openApiSpec` object, and then there's almost no need to have both the `getOAPath` function and `findOAPathMatchingPathname`, since almost everything is then moved inside the `findOAPathMatchingPathname` function.